Draft of release notes

agh1 opened the pull request:
Includes PRs merged through 999d52acff5fe81a2c7145f128cc622f50db02b3

([Working from this spreadsheet](https://github.com/civicrm/civicrm-core/files/610221/CiviCRM.4.7.14.Release.Notes.xlsx))

**Please note: this PR is a work-in-progress.**  I'm working one-by-one through the merged pull requests in chronological order.  I have not handled [civicrm-packages](https://github.com/civicrm/civicrm-packages), [civicrm-backdrop](https://github.com/civicrm/civicrm-backdrop), [civicrm-drupal](https://github.com/civicrm/civicrm-drupal), [civicrm-joomla](https://github.com/civicrm/civicrm-joomla), or [civicrm-wordpress](https://github.com/civicrm/civicrm-packages) yet either.

agh1 commented:
Some notes about the format:

* I would normally prefer outputting links at the bottom, but this file will eventually get very large, so scrolling to the bottom to view the thousandth link would be unwieldy.  Also, it would make it more difficult to generate the markdown from a spreadsheet.

* In the descriptions, I've tried to stick with the tone/tense of "this is how things do/should work *since the change*", which stands in contrast to Jira ticket descriptions, which typically say "this is how things work *prior to any improvement*"

agh1 commented on release-notes.md:
```diff
@@ -0,0 +1,299 @@
+
+This release was developed by the following code authors:
+
+AGH Strategies - Andrew Hunt; Australian Greens - Seamus Lee; CiviCRM - Coleman Watts, Jitendra Purohit, Monish Deb, Tim Otten, and Yashodha Chaku; CiviDesk - Nicolas Ganivet; CiviFirst - John Kirk; Coop SymbioTIC - Samuel Vanhove; Francesc Bassas i Bullich; Fuzion - Chris Burgess and Eileen McNaughton; Ginkgo Street Labs - Tobias Lounsbury; JMA Consulting - Edzel Lopez and Pradeep Nayak; Lightho ...
```
It's missing because I've only gotten through half of the PRs for this release.  I'll format it as such when I add you all along with the other contributors.

agh1 commented:
The release notes are complete as of f6102d194207401be23eef4db4549919cbab2b35.  I think what would make the most sense would be to merge this now (assuming it's acceptable) and then have later PRs to document last-minute additions to the release.  That would also allow others to propose changes to these notes via PR.

xurizaemon commented:
* Suggest moving changes above contributors
* Suggest renaming Bugs to Bugs resolved or similar, so it doesn't sound like "new bugs!" 😸

jamienovick commented:
@agh1 can we get Camilo Rodriguez marked as Compucorp organisation? Bit dissapointed we didnt have more names in there - I'll follow up and see why. Thanks

tttp commented:
@agh1 @scardinius is We Move Europe/Caltha

agh1 commented:
I moved the contributors to the bottom and added the folks you mentioned @jamienovick and @tttp. "Bugs" is now "Bugs resolved".  The section documenting the bugs introduced in this release is patch-welcome.

I marked a couple of issues as `[WIP]` or `[completes previous work]`.  For example, @colemanw pushed a commit that wiped out an issue that is incomplete.  The notes include all code changes, even if an issue isn't finished, but I can see how it would be confusing.  The actual notes (rather than the issue title) describe the changes that were made.  I force-pushed my change to wipe out @colemanw's commit that wiped out the note.

colemanw commented:
@agh1 that issue wasn't just incomplete, it was reverted out of this release entirely. So I don't think it belongs in the changelog this round.

colemanw commented:
This is great! Just two concerns:

How much time did it take to create this document? Seems like a lot of manual labor.

This document is rather long. I wonder if we ought to create one file per release instead of one monolithic file for every release?

agh1 commented:
I think this is all set (assuming no new changes in 4.7.14).  Please check the distmaker changes when you actually create the packaged files--I have had issues with testing distmaker in the past.

@colemanw it did take a lot of time creating this, but that's because it was all manual.  The plan is to automate much of it for future versions; I just wanted to get a feel for the process, and it all might be subject to change mid-stream while I figured it out.

I agree with your thought about one file per release: I created a folder called release-notes, put the main file as `release-notes/4.7.14.md`, and turned `release-notes.md` into a listing with links.
